packages/core/src/rum/instrumentation/DdRumUserInteractionTracking.tsx

```diff
@@ -37,27 +62,33 @@ export class DdRumUserInteractionTracking {
             return;
         }
         DdRumUserInteractionTracking.eventsInterceptor = new DdEventsInterceptor();
+
         const original = React.createElement;
         React.createElement = (
-            element: any,
-            props: any,
-            ...children: any
+            ...args: Parameters<typeof React.createElement>
         ): any => {
-            // check if we have an 'onPress' property and that this is really a function
-            if (props && typeof props.onPress === 'function') {
-                const originalOnPress = props.onPress;
-                props.onPress = (...args: any[]) => {
-                    DdRumUserInteractionTracking.eventsInterceptor.interceptOnPress(
-                        ...args
-                    );
-                    return originalOnPress(...args);
-                };
-                // we store the original onPress prop so we can keep memoization working
-                props.__DATADOG_INTERNAL_ORIGINAL_ON_PRESS__ = originalOnPress;
-            }
-            return original(element, props, ...children);
+            return this.patchCreateElementFunction(original, args);
         };
 
+        try {
+            // We have to use inline require here because older React versions (below 17) don't have jsx-runtime
+            // eslint-disable-next-line global-require, @typescript-eslint/no-var-requires
+            const jsxRuntime = require('react/jsx-runtime');
+            const originaljsx = jsxRuntime.jsx;
+            if (!originaljsx) {
```

Comment:
why do we need to throw an error instead of simply continuing? given that we have empty `catch` block for now anyway?

Reply:
There are some versions of React where `'react/jsx-runtime'` is exported but does not contain `jsx` yet. I would like to treat this cases the same as when it is not exported, so we can both log them into telemetry later so we can know how many clients are still running an old version of React.
And although it would work to continue, I believe it does not make a lot of sense if `jsx` is not exported.